app/src/main/java/com/example/poste/activities/FolderViewActivity.java

```diff
@@ -84,47 +91,56 @@ private void setListeners() {
 
     private void prepVars() {
         emptyText = findViewById(R.id.folderViewEmptyText);
+        emptyText.setVisibility(View.GONE); // Hide empty text until data is loaded
         folderName = findViewById(R.id.folderNameText);
+        folderName.setVisibility(View.INVISIBLE); // Hide folder name until data is loaded
+        // reset selected folder
+        User.getUser().setSelectedFolder(User.getUser().getFolder(User.getUser().getSelectedFolder().getId()));
+        // Set folder name in title bar
+        folderName.setText(User.getUser().getSelectedFolder().getTitle());
+        folderName.setVisibility(View.VISIBLE); // Show folder name now that data is loaded
         newBut = findViewById(R.id.newPost);
         settingsBut = findViewById(R.id.folderSettings);
         postRecyclerView = findViewById(R.id.posts_recycler_view);
+        postAdapter = new PostAdapter(
+                new PostAdapter.ClickListener() {
+                    @Override
+                    public void onItemClick(int position, Post post) {
+                        User.getUser().setSelectedPost(post);
+                        try {
+                            Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(User.getUser().getSelectedPost().getUrl()));
+                            startActivity(browserIntent);
+                        } catch (Exception e) {
+                            e.printStackTrace();
+                            Toast.makeText(FolderViewActivity.this, R.string.internal_error, Toast.LENGTH_LONG).show();
+                        }
+                    }
+
+                    @Override
+                    public void onItemLongClick(int position, Post post) {
+                        User.getUser().setSelectedPost(post);
+                    }
+                },
+                User.getUser().getSelectedFolder().getPosts()
+        );
+        postRecyclerView.setAdapter(postAdapter);
+        postRecyclerView.setLayoutManager(new LinearLayoutManager(FolderViewActivity.this));
```

Comment:
Moved all of this into prepVars for consistency (and to avoid calling it numerous times)